Add docs for re-enabling FIPS after Strimzi 0.33 upgrade (#7880)
strimzi/strimzi-kafka-operator: documentation/modules/upgrading/proc-reenabling-FIPS-mode.adoc

// This module is included in the following assemblies:
//
// assembly-upgrade.adoc

[id='proc-reenabling-FIPS-mode-{context}']
= Re-enabling FIPS mode

[role="_abstract"]

> Until Strimzi 0.33, running on FIPS-enabled Apache Kafka clusters was possible only with the FIPS mode disabled using the `FIPS_MODE` environment variable.
> But from 0.33 on, Strimzi supports running on FIPS-enabled Kubernetes clusters without disabling the FIPS mode.
> If you run Strimzi on a FIPS-enabled Kubernetes cluster with the `FIPS_MODE` set to `disabled`, you can re-enable it by following this procedure.

Reviewer: If you have 
```suggestion
Upgrade Strimzi to run in FIPS mode on FIPS-enabled Kubernetes clusters. 
Until Strimzi 0.33, running on FIPS-enabled Kubernetes clusters was possible only by disabling FIPS mode using the `FIPS_MODE` environment variable.
From release 0.33, Strimzi supports FIPS mode.
If you run Strimzi on a FIPS-enabled Kubernetes cluster with the `FIPS_MODE` set to `disabled`, enable it by following this procedure.
```

.Prerequisites

* FIPS-enabled Kubernetes cluster
* An existing Cluster Operator deployment with the `FIPS_MODE` environment variable set to `disabled`

.Procedure

> . Upgrade Strimzi Cluster Operator to version 0.33 or newer but keep the `FIPS_MODE` environment variable set to `disabled`

Reviewer: ```suggestion
. Upgrade the Cluster Operator to version 0.33 or newer but keep the `FIPS_MODE` environment variable set to `disabled`.
```

> . If you initially deployed your cluster with a Strimzi version older than 0.30, it might still use the PKCS12 stores using old encryption and digest algorithms which are not supported with FIPS enabled.

Reviewer: ```suggestion
. If you initially deployed a Strimzi version older than 0.30, it might use old encryption and digest algorithms in its PKCS #12 stores, which are not supported with FIPS enabled.
```

>   To get the certificates recreated with the new algorithms, renew the Cluster and Clients CA.

Reviewer: ```suggestion
  To recreate the certificates with updated algorithms, renew the Cluster and Clients CA.
```

> .. For renewing the CAs generated by the Cluster Operator, use the link:{BookURLUsing}#proc-renewing-ca-certs-manually-str[annotation to trigger the renewal^].
> 
> .. For renewing your own CAs, follow the docs about link:{BookURLUsing}#renewing-your-own-ca-certificates-str[renewing your own CA certificates^].

Reviewer: ```suggestion
.. To renew the CAs generated by the Cluster Operator, link:{BookURLUsing}#proc-renewing-ca-certs-manually-str[add the `force-renew` annotation to the CA secret to trigger a renewal^].

.. To renew your own CAs, link:{BookURLUsing}#renewing-your-own-ca-certificates-str[add the certificate to the CA secret and update the `ca-cert-generation` annotation with a higher incremental value to capture the update^].
```

> . If you use SCRAM-SHA authentication, you have to check the password length of your users.

Reviewer: ```suggestion
. If you use SCRAM-SHA-512 authentication, check the password length of your users.
```

>   If they are less than 32 characters long, you will have to generate a new password for them.

Reviewer: ```suggestion
  If they are less than 32 characters long, generate a new password in one of the following ways:
```

> .. You can do that by deleting the user secret and having the User Operator generate a new one with a new password with sufficient length.

Reviewer: ```suggestion
.. Delete the user secret so that the User Operator generates a new one with a new password of sufficient length.
```

> .. Alternatively, if you provided your password using the `.spec.authentication.password` section of the `KafkaUser` custom resource, you have to update the password in the Kubernetes Secret referenced here.

Reviewer: ```suggestion
.. If you provided your password using the `.spec.authentication.password` properties of the `KafkaUser` custom resource, update the password in the Kubernetes secret referenced in the same password configuration.
```

   Don't forget to update your clients to use the new passwords.

> . Once all the certificates are using the correct algorithms and the SCRAM-SHA passwords are sufficiently long, you can now enable the FIPS mode.

Reviewer: ```suggestion
+
When all the certificates are using the correct algorithms and the SCRAM-SHA-512 passwords are sufficiently long, you can enable the FIPS mode.
```

Reviewer: I'd add this as a verification to step 3 before moving onto the next step.

>   You can do so by removing the environment variable `FIPS_MODE` from the Strimzi Cluster Operator deployment.
>   The operator will restart and will roll all the operands to re-enable the FIPS mode.
>   And once it is done, all your Kafka clusters will not run with the OpenJDK FIPS mode enabled.

Reviewer: ```suggestion
. Remove the `FIPS_MODE` environment variable from the Cluster Operator deployment.
+
You can use the `060-Deployment-strimzi-cluster-operator.yaml` installation file to make this change.

. Deploy the Cluster Operator:
+
[source,shell]
kubectl create -f install/cluster-operator -n myproject
+
This restarts the Cluster Operator and rolls all the operands to enable the FIPS mode.
After the restart is complete, all Kafka clusters now run with FIPS mode enabled.
```

Reviewer: It said _not run_ here. I don't know if I'm reading the procedure incorrectly, but I'm assuming they can now run in FIPS mode where appropriate.

Author: I mostly followed the suggestion here. But left out the part about the exact files since that would not cover the other installation methods. So I think it is better to leave it to the users to resolve it in whatever way they used for installation.
